File: src/main/java/com/hemanth/mockito/test_doubles/spy/BookService.java
package com.hemanth.mockito.test_doubles.spy;

public class BookService {

    private BookRespository bookRespository;

    public BookService(BookRespository bookRespository) {
        this.bookRespository = bookRespository;
    }

    public void addBook(Book book){
        if (book.getPrice() > 400){
            return;
        }
        bookRespository.save(book);
    }

}

File: src/main/java/com/hemanth/mockito/behavior/verification/BookService.kt
package com.hemanth.mockito.behavior.verification

import com.hemanth.mockito.annotations.support.BookRespository

class BookService(val bookRespository: BookRepository) {

    fun addBook(book: Book) {
        bookRespository.save(book)
    }

    fun addBook(bookRequest: BookRequest) {
        if (bookRequest.price <= 500) {
            return
        }
        var book = Book(
            null,
            bookRequest.title,
            bookRequest.price,
            bookRequest.publishedDate
        )
        bookRespository.save(book)
    }

    fun updatePrice(updatedPrice: Int, bookId: String?){
        if (bookId == null){
            return
        }
        var book:Book = bookRespository.findBookByBookId(bookId)
       /* if (book.price == updatedPrice){
            return
        }*/
        book.price = updatedPrice
        bookRespository.save(book)
    }

}